baseapp/proxy_writer.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package baseapp
+
+import (
+	"bufio"
+	"io"
+	"net"
+	"net/http"
+)
+
+// ProxyResponseWriter is a proxy around an http.ResponseWriter
+// that counts bytes written to the wrapped ResponseWriter.
+type ProxyResponseWriter interface {
```

Comment:
Maybe `CountingResponseWriter` or `RecordingResponseWriter`? Proxy is kind of an overloaded term in network applications, especially for an exported type.

Reply:
`RecordingResponseWriter`